src/main/kotlin/org/jetbrains/tinygoplugin/services/ErrorRegexUtil.kt

```diff
@@ -0,0 +1,71 @@
+package org.jetbrains.tinygoplugin.services
+
+import com.goide.sdk.GoSdkService
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import org.jetbrains.tinygoplugin.TinyGoBundle
+
+private const val TINYGO_INCOMPATIBLE_GO_VERSION_UNKNOWN = "tinygoSDK.incompatibleGoVersion.unknownVersion"
+private const val TINYGO_INCOMPATIBLE_GO_VERSION_TITLE = "tinygoSDK.incompatibleGoVersion.title"
+private const val TINYGO_INCOMPATIBLE_GO_VERSION_RANGE_SPECIFIED_MESSAGE =
+    "tinygoSDK.incompatibleGoVersion.rangeSpecified.message"
+private const val TINYGO_INCOMPATIBLE_GO_VERSION_RANGE_UNSPECIFIED_MESSAGE =
+    "tinygoSDK.incompatibleGoVersion.rangeUnspecified.message"
+
+fun generateMessageIfVersionErrorFound(project: Project, processOutput: String): String? {
+    val incompatibleGoVersionsMatch = extractIncompatibleVersionsError(processOutput)
+    val couldNotReadGoVersionMatch = extractCouldNotReadGoVersionError(processOutput)
+    val errorMessage =
+        if (incompatibleGoVersionsMatch != null) {
+            val (currentVer, oldestVer, latestVer) = incompatibleGoVersionsMatch
+            TinyGoBundle.message(
+                TINYGO_INCOMPATIBLE_GO_VERSION_RANGE_SPECIFIED_MESSAGE,
+                currentVer,
+                oldestVer,
+                latestVer
+            )
+        } else if (couldNotReadGoVersionMatch) {
+            val goSdk = GoSdkService.getInstance(project).getSdk(null)
+            val currentVer = goSdk.version ?: TinyGoBundle.message(TINYGO_INCOMPATIBLE_GO_VERSION_UNKNOWN)
+            TinyGoBundle.message(
+                TINYGO_INCOMPATIBLE_GO_VERSION_RANGE_UNSPECIFIED_MESSAGE,
+                currentVer
+            )
+        } else null
+
+    if (errorMessage != null) {
+        Messages.showErrorDialog(
+            project,
+            errorMessage,
+            TinyGoBundle.message(TINYGO_INCOMPATIBLE_GO_VERSION_TITLE)
+        )
+    }
+
+    return errorMessage
+}
+
+@Suppress("MagicNumber")
+fun extractIncompatibleVersionsError(msg: String): Triple<String, String, String>? {
+    val minor2DigitsRule = "([0-9]\\.[0-9][0-9])"
+    val minor1DigitRule = "([0-9]\\.[0-9])" // rule produces 3 groups: (1d or 2d), (1d), (2d)
+    val goVersionRule = "($minor1DigitRule|$minor2DigitsRule)"
+    val errorPattern =
+        Regex("""requires go version $goVersionRule through $goVersionRule, got go$goVersionRule\n""")
+
+    if (!errorPattern.containsMatchIn(msg)) {
```

Comment:
We use the pattern twice in the case when it finds something. It's not critical but can be inproved.